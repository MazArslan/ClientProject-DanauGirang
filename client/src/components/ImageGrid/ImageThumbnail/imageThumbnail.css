.card-heading {
	margin-top: 6px;
	margin-left: 6px;
}

.dot-true {
	height: 25px;
	width: 25px;
	background-color: #99ff66;
	border-radius: 50%;
	display: inline-block;
}

.dot-false {
	height: 25px;
	width: 25px;
	background-color: #ff6666;
	border-radius: 50%;
	display: inline-block;
}
