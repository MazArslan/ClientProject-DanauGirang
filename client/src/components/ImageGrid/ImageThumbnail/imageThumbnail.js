import React from "react";
import propTypes from "prop-types";
import { Row, Col, Card, CardImg } from "reactstrap";
import "./imageThumbnail.css";

/**
 * ImageThumbnail component
 * Converts the image binary of an image and  displays it as a thumbnail in a card
 * @param { id, image, cardId, dateTime, isComplete } props
 */
const imageThumbnail = props => {
  const { id, image, cardId, dateTime, isComplete } = props;

  const date = String(dateTime).substr(0, 10);

  return (
    <React.Fragment>
      <Card className="card">
        <div className="card-heading">
          <Row>
            <Col xs="2">
              <span className={`dot-${isComplete}`}></span>
            </Col>
            <Col xs="4">
              <b>{`Image ${cardId}`}</b>
            </Col>
            <Col sm={{ size: "auto", offset: 1 }}>
              <p>{date}</p>
            </Col>
          </Row>
        </div>
        <CardImg
          top
          width="20%"
          src={`/api/images/cameraTrap/${image.filePath}`}
          alt={`${id}`}
        />
      </Card>
    </React.Fragment>
  );
};

imageThumbnail.propTypes = {
  id: propTypes.string,
  image: propTypes.object.isRequired,
  cardId: propTypes.number,
  dateTime: propTypes.string,
  isComplete: propTypes.bool
};

export default imageThumbnail;
