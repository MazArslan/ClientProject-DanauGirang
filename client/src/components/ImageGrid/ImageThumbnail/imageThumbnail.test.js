import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import ImageThumbnail from "./imageThumbnail";

Enzyme.configure({ adapter: new Adapter() });

describe("<ImageThumbnail/>", () => {
  const props = {
    id: "Unique id string",
    image: {
      filePath: "5e123ty34u35567h2g/0001.png"
    },
    cardId: 2,
    dateTime: "2020-03-22T01:00:00.000+00:00",
    isComplete: false
  };
  const container = shallow(<ImageThumbnail {...props} />);

  describe("component rendering", () => {
    it("should render correctly", () => {
      expect(container.exists()).toBe(true);
      expect(toJson(container)).toMatchSnapshot();
    });

    it("should render a completion marker", () => {
      expect(container.find("span").props().className).toEqual("dot-false");
    });

    it("should render a heading", () => {
      expect(container.find("b").props().children).toEqual("Image 2");
    });

    it("should render an image", () => {
      const imageString = "/api/images/cameraTrap/5e123ty34u35567h2g/0001.png";
      expect(container.find("CardImg").props().src).toEqual(imageString);
    });
  });
});
