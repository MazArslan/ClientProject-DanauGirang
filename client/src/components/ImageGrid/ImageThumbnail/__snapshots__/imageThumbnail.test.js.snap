// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageThumbnail/> component rendering should render correctly 1`] = `
<Fragment>
  <Card
    className="card"
    tag="div"
  >
    <div
      className="card-heading"
    >
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="2"
        >
          <span
            className="dot-false"
          />
        </Col>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="4"
        >
          <b>
            Image 2
          </b>
        </Col>
        <Col
          sm={
            Object {
              "offset": 1,
              "size": "auto",
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p>
            2020-03-22
          </p>
        </Col>
      </Row>
    </div>
    <CardImg
      alt="Unique id string"
      src="/api/images/cameraTrap/5e123ty34u35567h2g/0001.png"
      tag="img"
      top={true}
      width="20%"
    />
  </Card>
</Fragment>
`;
